spec/offline_sort/offline_sort_spec.rb

```diff
@@ -1,113 +1,138 @@
 require 'spec_helper'
 
 describe OfflineSort::Sorter do
+  describe "#sort" do
+    subject { offline_sorter_instance.sort.to_a }
 
-  shared_examples "a correct offline sort" do
-    let(:count) { 10000 }
-    let(:entries_per_chunk) { 900 }
-    let(:enumerable) {}
-    let(:sort) {}
-    let(:unsorted) { enumerable.dup }
-
-    subject do
-      OfflineSort.sort(enumerable, chunk_size: entries_per_chunk, &sort)
+    let(:offline_sorter_instance) do
+      described_class.new(enumerable, chunk_size: entries_per_chunk, &sort)
     end
 
-    it "writes out to disk" do
-      expect(Tempfile).to receive(:open).at_least(:once).and_call_original
-      subject
+    let(:entries_per_chunk) { 900 }
+    let(:count) { 10000 }
+    let(:enumerable) { arrays }
+    let(:sort) { array_sort }
+    let(:array_sort) { Proc.new { |arr| arr[2] } }
+    let(:arrays) do
+      count.times.map do |index|
+        [SecureRandom.hex, index, SecureRandom.hex]
+      end
     end
 
-    shared_examples "produces a sorted result" do
-      it "produces the same sorted result as an in-memory sort" do
-        sorted = subject.to_a
+    shared_examples "a correct offline sort" do
+      let(:unsorted) { enumerable.dup }
+
+      it "writes out to disk" do
+        expect(Tempfile).to receive(:open).at_least(:once).and_call_original
+        subject
+      end
+
+      shared_examples "produces a sorted result" do
+        it "produces the same sorted result as an in-memory sort" do
+          expect(unsorted).to match_array(enumerable)
+          expect do
```

Comment:
Since the input to this is now randomly generated, is it worth still having the extra logic to print out where the difference was? Could we just have this be `expect(unsorted.sort_by(&sort)).to eq(sorted)`?